designs/lily/src/front.mjs

```diff
@@ -0,0 +1,610 @@
+   */
+  const adaptInseamAndOutseam = () => {
+    let shift = [
+      'kneeInCp1',
+      'kneeInCp2',
+      'kneeOutCp1',
+      'kneeOutCp2',
+      'kneeIn',
+      'kneeOut',
+      'knee',
+      'floorInCp2',
+      'floorIn',
+      'floorOutCp2',
+      'floorOut',
+      'floor',
+      'grainlineBottom',
+    ]
+    let delta = seamDelta()
+    let run = 0
+    do {
+      run++
+      for (const i of shift) points[i] = points[i].shift(90, delta)
+      delta = seamDelta()
+    } while (Math.abs(delta) > 1 && run < 10)
+  }
+  /*
+   * Helper method to determine the delta common when both inseam and outseam
+   * are either too long or too short
+   */
+  const seamDelta = () => {
+    let inseam = inseamDelta()
+    let outseam = outseamDelta()
+    return Math.abs(inseam) > Math.abs(outseam) ? outseam : inseam
+  }
+  /*
+   * Helper method that can fit either inseam or outseam
+   */
+  const adaptSeam = (side) => {
+    const out = side === 'out' ? true : false
+    let rotate = [
+      'cfSeat',
+      'crotchSeamCurveCp1',
+      'crotchSeamCurveCp2',
+      'crotchSeamCurveStart',
+      'waistIn',
+      'cfWaist',
+      'waistOut',
+    ]
+    rotate.push(out ? 'seatOut' : 'fork')
+    const deltaMethod = out ? outseamDelta : inseamDelta
+    let run = 0
+    let delta = deltaMethod()
+    do {
+      for (const i of rotate)
+        points[i] = points[i].rotate(
+          (delta / 10) * (out ? 1 : -1),
+          points[out ? 'fork' : 'seatOut']
+        )
+      run++
+      delta = deltaMethod()
+      console.log('delta', delta)
+    } while (Math.abs(delta) > 1 && run < 20)
+    console.log('run', run)
+  }
+  const adaptOutseam = () => adaptSeam('out')
+  const adaptInseam = () => adaptSeam('in')
+
+  console.log('test')
+
+  // NOTE: majority of points re-used from titan
+
+  // shape at the ankle (unlike titan)
+  let halfAnkle = store.get('halfAnkle')
+  // let halfAnkle = store.get('kneeFront') / 2
+
+  points.floorOut = points.floor.shift(180, halfAnkle)
+  points.floorIn = points.floorOut.flipX(points.floor)
+
+  console.log('ankle')
+
+  // Control points between knee and ankle
+  points.floorInCp2 = points.floorIn.shift(90, points.knee.dy(points.floor) / 3)
+  points.kneeInCp1 = points.kneeIn.shift(90, -points.knee.dy(points.floor) / 3)
+  points.floorOutCp2 = points.floorOut.shift(90, points.knee.dy(points.floor) / 3)
+  points.kneeOutCp2 = points.kneeOut.shift(90, -points.knee.dy(points.floor) / 3)
+
+  // other control points have already been calculated in titan
+  // Control points to shape the legs towards the seat
+  // Balance the waist
+
+  // Draw initial crotch seam
+  drawCrotchSeam()
+
+  // Uncomment this to see the outline prior to fitting the crotch seam
+  //paths.seam1 = drawPath().attr('class', 'dashed lining')
+
+  if (options.fitCrossSeam && options.fitCrossSeamFront) {
+    let delta = crotchSeamDelta()
+    let rotate = ['waistIn', 'waistOut', 'cfWaist']
```

Comment:
Use a `const` here

Reply:
As a matter of fact, since it's only 3 entries, just inline it in the for loop

Reply:
Could you show me how to do that?